docs/development/pigs/pig-004.rst

```diff
@@ -0,0 +1,131 @@
+stable datasets in tutorials to mostly save the versioning issues. To ship
+tutorials and notebooks to users, we propose to add a ``gammapy download``
+command. ``gammapy-extra`` will remain as a repository for developers and as one
+place where datasets can be put, but it will not be mentioned to users.
+
+What we have
+============
+
+We have the `gammapy`_ repository for code, and the `gammapy-extra`_ repository
+for tutorial notebooks, example datasets and a few other things.
+
+The ``gammapy`` repository currently is 11 MB and the ``gammapy-extra`` repository
+is 157 MB. In ``gammapy-extra/notebooks``, we have ~ 10 tutorial notebooks, each
+20 kB to 1 MB in size, i.e. a few MB in total. Most of the size comes from PNG
+output images in the notebooks, and they usually change on re-run, i.e. even
+though git compresses a bit, the repo grows by up to 1 MB every time a notebook
+is edited. The data files we access from the tutorials are maybe 20 or 30 MB, a
+lot of the datasets we have there are old and should be cleaned up / removed.
+The reason the notebooks and data files were split out was to keep the code
+repository small and avoid it growing to 100 MB or even 1 GB over time.
+
+This separation of code and notebooks and datasets has been a problem in Gammapy
+for years.
+
+Given that Gammapy code changes (and probably always will, even if the pace will
+become slower and slower over the years), the tutorials have to be
+version-coupled to the code. A related question is how tutorial notebooks and
+example datasets are shipped to users. Possible solutions to this and related
+problems have been discussed serval times, see e.g. `GH 1237`_, `GH 1369`_, `GH
+700`_, `GH 431`_, `GH 405`_, `GH 228`_ (probably missed a few).
+
+
+Proposal
+========
+
+This proposal is limited. It outlines a few key changes in the setup for
+notebooks and example data that mostly solve the versioning and shipping issue.
+It does not outline e.g. complete lists of notebooks and datasets, those will be
+iteratively improved in the coming month, with or without an extra PIG.
+
+To solve the versioning issue for notebooks, we propose to move the notebooks
+from ``gammapy-extra/notebooks`` to ``gammapy/notebooks``. Whether to store
+notebooks with or without output cells filled in the repository is a difficult
+decision. Removing output cells before committing has the advantage that the
+files are small, and that the diff in the pull request is also small and review
+becomes possible. On the other hand, the output is not directly visible on
+Github. Note that in any case, a rendered version of the notebooks will be
+available via the docs, that is already in place. We propose to remove the
+output and try that workflow for a while, see if we like it. Since notebooks
+aren't very large, and updates are not needed very often, we think changing to
+notebooks with output would also be acceptable in the ``gammapy`` repository.
+
```

Comment:
It is probably more natural to look at notebooks from the docs. It therefore seems reasonable not to commit the output.

Is there a way to prevent someone from committing notebooks with output cells?

Reply:
> Is there a way to prevent someone from committing notebooks with output cells?

Developer documentation and code review. Let's not put effort into more automation like git commit hooks or bots that review PRs. Those things we might have in the future if we get an engineer at some point, but for now it's overkill.